File: src/main/java/com/anfema/ionclient/utils/IonLog.java
package com.anfema.ionclient.utils;

import android.util.Log;

import com.anfema.ionclient.IonConfig;

public class IonLog
{
	public static final int VERBOSE = Log.VERBOSE;
	public static final int DEBUG   = Log.DEBUG;
	public static final int INFO    = Log.INFO;
	public static final int WARN    = Log.WARN;
	public static final int ERROR   = Log.ERROR;
	public static final int NONE    = Integer.MAX_VALUE;

	private static String defaultTag = "IonClient";

	// prevent initialization
	private IonLog()
	{
	}

	public static void v( String tag, String message )
	{
		if ( IonConfig.logLevel == VERBOSE )
		{
			Log.v( tag, message );
		}
	}

	public static void v( String message )
	{
		v( defaultTag, message );
	}

	public static void d( String tag, String message )
	{
		if ( IonConfig.logLevel <= DEBUG && IonConfig.logLevel >= VERBOSE )
		{
			Log.d( tag, message );
		}
	}

	public static void d( String message )
	{
		d( defaultTag, message );
	}

	public static void i( String tag, String message )
	{
		if ( IonConfig.logLevel <= INFO && IonConfig.logLevel >= VERBOSE )
		{
			Log.i( tag, message );
		}
	}

	public static void i( String message )
	{
		i( defaultTag, message );
	}

	public static void w( String tag, String message )
	{
		if ( IonConfig.logLevel <= WARN && IonConfig.logLevel >= VERBOSE )
		{
			Log.w( tag, message );
		}
	}

	public static void w( String message )
	{
		w( defaultTag, message );
	}

	public static void e( String tag, String message )
	{
		if ( IonConfig.logLevel <= ERROR && IonConfig.logLevel >= VERBOSE )
		{
			Log.e( tag, message );
		}
	}

	public static void e( String message )
	{
		e( defaultTag, message );
	}

	public static void ex( String tag, Throwable exception )
	{
		if ( IonConfig.logLevel <= ERROR && IonConfig.logLevel >= VERBOSE )
		{
			Log.e( tag, Log.getStackTraceString( exception ) );
		}
	}

	public static void ex( Throwable exception )
	{
		ex( defaultTag, exception );
	}
}

File: src/main/java/com/anfema/ionclient/utils/IonLog.kt
package com.anfema.ionclient.utils

import android.util.Log

object IonLog {

    const val VERBOSE = Log.VERBOSE
    const val DEBUG = Log.DEBUG
    const val INFO = Log.INFO
    const val WARN = Log.WARN
    const val ERROR = Log.ERROR
    const val NONE = Int.MAX_VALUE

    /** Set log level for ION client */
    var logLevel = NONE

    /** set the default Tag used when functions are called, where no log tag is provided */
    var defaultTag = "IonClient"

    @JvmStatic
    fun v(tag: String, message: String) {
        if (logLevel == VERBOSE) {
            Log.v(tag, message)
        }
    }

    @JvmStatic
    fun v(message: String) {
        v(defaultTag, message)
    }

    @JvmStatic
    fun d(tag: String, message: String) {
        if (logLevel in VERBOSE..DEBUG) {
            Log.d(tag, message)
        }
    }

    @JvmStatic
    fun d(message: String) {
        d(defaultTag, message)
    }

    @JvmStatic
    fun i(tag: String, message: String) {
        if (logLevel in VERBOSE..INFO) {
            Log.i(tag, message)
        }
    }

    @JvmStatic
    fun i(message: String) {
        i(defaultTag, message)
    }

    @JvmStatic
    fun w(tag: String, message: String) {
        if (logLevel in VERBOSE..WARN) {
            Log.w(tag, message)
        }
    }

    @JvmStatic
    fun w(message: String) {
        w(defaultTag, message)
    }

    @JvmStatic
    fun e(tag: String, message: String) {
        if (logLevel in VERBOSE..ERROR) {
            Log.e(tag, message)
        }
    }

    @JvmStatic
    fun e(message: String) {
        e(defaultTag, message)
    }

    @JvmStatic
    fun ex(tag: String, exception: Throwable) {
        if (logLevel in VERBOSE..ERROR) {
            Log.e(tag, Log.getStackTraceString(exception))
        }
    }

    @JvmStatic
    fun ex(exception: Throwable) {
        ex(defaultTag, exception)
    }
}
